src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1052,7 +1052,10 @@ void JavaThread::set_exception_oop(oop o) {
 }
 
 void JavaThread::handle_special_runtime_exit_condition() {
-  if (is_obj_deopt_suspend()) {
+  // We mustn't block for object deopt if the thread is
+  // currently executing in a JNI critical region, as that
+  // can cause deadlock because the GCLocker is held.
```

Comment:
I think the last part of the comment might be confusing because it's the deopt suspender that helds the lock waiting for this thread to exit the critical region [[1]](https://github.com/openjdk/jdk/blob/0b2712400b55d4a512db225d090c2f06f01f7f1f/src/hotspot/share/gc/shared/gcLocker.cpp#L108). Maybe "... cause deadlock because the suspender may allocate and block waiting for this thread to exit the critical region"?